docs/source/metadata_parsing.mdx

```diff
@@ -92,40 +92,41 @@ export type SafetensorsShardedHeaders = Record<FileName, SafetensorsFileHeader>;
 
 ### Python
 
-In this example python script, we are parsing metadata of [gpt2](https://huggingface.co/gpt2/blob/main/model.safetensors).
+[`huggingface_hub`](https://huggingface.co/docs/huggingface_hub) provides a Python API to parse safetensors metadata.
+Use [`get_safetensors_metadata`](https://huggingface.co/docs/huggingface_hub/package_reference/hf_api#huggingface_hub.HfApi.get_safetensors_metadata) to get all safetensors metadata of a model.
+Depending on if the model is sharded or not, one or multiple safetensors files will be parsed.
 
 ```python
-import requests # pip install requests
-import struct
-
-def parse_single_file(url):
-    # Fetch the first 8 bytes of the file
-    headers = {'Range': 'bytes=0-7'}
-    response = requests.get(url, headers=headers)
-    # Interpret the bytes as a little-endian unsigned 64-bit integer
-    length_of_header = struct.unpack('<Q', response.content)[0]
-    # Fetch length_of_header bytes starting from the 9th byte
-    headers = {'Range': f'bytes=8-{7 + length_of_header}'}
-    response = requests.get(url, headers=headers)
-    # Interpret the response as a JSON object
-    header = response.json()
-    return header
-
-url = "https://huggingface.co/gpt2/resolve/main/model.safetensors"
-header = parse_single_file(url)
```

Comment:
i think this was still valuable to show how it actually works, no? potentially mentioning the `huggingface_hub` support at the end

Reply:
Linking @mishig25's PR: https://github.com/huggingface/safetensors/pull/424 (thanks about that :pray:)